### DISCUSSION TITLE
文章权限 用户管理

### DISCUSSION THREAD

**Discussion by kaizcc** [halo-dev/halo#6745] [Ideas]
### 你当前使用的版本

2.19

### 描述一下此特性

我已经在使用文章限制阅读的插件了，可是有些功能还是无法实现。是否可以通过原生态用户角色或者插件实现这种功能呢：
1. 部分文章，页面需要不同的角色才能访问。这样就可以将用户进行管理
2. 是否可以支持用户付费转换角色的功能，从而实现类似WordPress会员制内容管理。
感谢回答！谢谢

**Comment by ruibaby** [comment]
考虑过通过插件实现类似的功能，但目前核心成员精力有限，不能提供具体开发计划。

可以看看社区是否有开发者愿意实现这样的插件。

**Reply by AffiNiNy** [reply]
> 考虑过通过插件实现类似的功能，但目前核心成员精力有限，不能提供具体开发计划。
> 
> 可以看看社区是否有开发者愿意实现这样的插件。

目前对插件工程(plugin-starter)研究过一翻. 如果是通过插件来实现角色与文章, 角色与文章分类的权限绑定, 有什么思路实现吗?

我已阅读过[WebFilter拓展点](https://docs.halo.run/2.20/developer-guide/plugin/extension-points/server/additional-webfilter), webfilter 过滤不了 Post Category tag 这些关键类.

又尝试过包装 ReactiveExtensionClient 接口类, 但这个接口没有继承 ExtensionPoint, 没有做成拓展点. 否则就可以改造 ReactiveExtensionClient 的方法了.

纯靠插件来处理可行吗?

**Reply by liuyiwuqing** [reply]
通过插件可行，目前我已经通过插件实现了
https://blog.muyin.site/archives/Y15zB9rk

@kaizcc @AffiNiNy @CF673

**Comment by CF673** [comment]
内置原生密码访问，评论查看，这是必需品，居然单独收费，说不过去，还是接着用wp ty吧……

**Reply by ruibaby** [reply]
继续用吧。
